StartBonusActivation/handler.ts

```diff
@@ -369,13 +335,13 @@ export function StartBonusActivationHandler(
   eligibilityCheckModel: EligibilityCheckModel
 ): IStartBonusActivationHandler {
   return async (context, fiscalCode) => {
-    const client = df.getClient(context);
+    const dfClient = df.getClient(context);
 
     return taskEither
       .of<StartBonusActivationResponse, void>(void 0)
-      .chain(_ => checkEligibilityCheckIsRunning(client, fiscalCode))
-      .chain(_ => checkBonusActivationIsRunning(client, fiscalCode))
-      .chain(_ => getLatestValidDSU(eligibilityCheckModel, fiscalCode))
+      .chainSecond(checkEligibilityCheckIsRunning(dfClient, fiscalCode))
```

Comment:
Could we handle the response type at the caller level instead than inside the method `checkEligibilityCheckIsRunning`?
We can handle it like this
```suggestion
      .chainSecond(checkEligibilityCheckIsRunning(dfClient, fiscalCode))
      .mapLeft<StartBonusActivationResponse>(_ =>
        ResponseErrorInternal("Error")
      )
      .chain(
        fromPredicate(
          _ => _ === false,
          () => ResponseErrorForbiddenNotAuthorized
        )
      )
```

Reply:
isn't _ => _ === true ?

Reply:
No, the predicate is the right path condition. So, if `checkEligibilityCheckIsRunning` returns true the `taskEither` will be `left(ResponseErrorForbiddenNotAuthorized)`.

Reply:
I like the idea that this check doesn't introduce noise in the handler pipeline. The idea is: you know this would procede in the left path if the check is negative, and that's all you need to understand its meaning. If you need to know exactly which kind of failures it can generate, you can dig more.

Reply:
ok I've considered this which is a good suggestion but the fact is that all the methods in the chain handle responses internally so this would require a huge refactor of the whole module :)

Reply:
👍